addons/account/models/account_move.py

```diff
@@ -1892,6 +1899,18 @@ def _compute_taxes_legal_notes(self):
                 if not is_html_empty(tax.invoice_legal_notes)
             )
 
+    @api.depends('line_ids')
```

Comment:
```suggestion
    @api.depends('line_ids.payment_date')
```
The depends will trigger on new/delete lines, but not if one of the subfields is modified if I'm not mistaken :thinking:

Reply:
Also added line_ids.reconciled but yeah, it's indeed better